tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -2230,7 +2230,33 @@ Status ConvertConv2DHelper(OpConverterParams* params, int group,
     conv_layer = layer;
   }
   nvinfer1::ITensor* output_tensor = conv_layer->getOutput(0);
-
+  // Add an extra padding for Deconv because TRT doesn't accept the
+  // argument output_shape and thus the TRT output shape could be wrong
+  // in case of strides>1.
+  if (is_conv2d_backprop_input) {
+    auto tf_output_shape = backprop_output_size.GetTrtDims();
+    nvinfer1::Dims trt_output_shape = output_tensor->getDimensions();
+    // What determines the padding size is the difference between the given
+    // input_sizes (tf_output_shape) and TRT computed size.
+    const int height_diff = tf_output_shape.d[h_index - 1] - trt_output_shape.d[1];
```

Comment:
Do we need to consider NHWC vs NCHW to determine whether to use trt_output_shape.d[1] or d[2]? Same below.

Reply:
It seems the output is always NCHW? Then should it be d[2] instead?

Reply:
`trt_output_shape` is the output of TRT which is always NCHW.